ethpm/backends/registry.py

```diff
@@ -104,15 +107,35 @@ def parse_registry_uri(uri: str) -> RegistryURI:
         raise CannotHandleURI(
             f"Invalid address or ENS domain found in uri: {uri}."
         )
-    pkg_name, pkg_version = _process_pkg_path(parsed_uri.path)
-    return RegistryURI(address, chain_id, pkg_name, pkg_version, ens)
+    pkg_name, pkg_version, namespaced_asset = _process_pkg_path(parsed_uri.path)
+    return RegistryURI(address, chain_id, pkg_name, pkg_version, namespaced_asset, ens)
+
 
+def _process_pkg_path(raw_pkg_path: str) -> Tuple[Optional[str], Optional[str], Optional[str]]:
+    pkg_path = raw_pkg_path.strip("/")
+    if not pkg_path:
+        return None, None, None
+
+    pkg_id, namespaced_asset = _parse_pkg_path(pkg_path)
+    pkg_name, pkg_version = _parse_pkg_id(pkg_id)
+    if not pkg_version and namespaced_asset:
+        raise EthPMValidationError(
+            "Invalid registry URI, missing package version."
+            "Version is required if namespaced assets are defined."
+        )
+    return pkg_name, pkg_version, namespaced_asset
+
+
+def _parse_pkg_path(pkg_path: str) -> Tuple[str, Optional[str]]:
+    if "/" in pkg_path:
+        pkg_id = pkg_path.split("/")[0]
+        namespaced_asset = "/".join(pkg_path.split("/")[1:])
+        return pkg_id, namespaced_asset
+    else:
+        return pkg_path, None
 
-def _process_pkg_path(pkg_path: str) -> Tuple[Optional[str], Optional[str]]:
-    pkg_id = pkg_path.strip("/")
-    if not pkg_id:
-        return None, None
 
+def _parse_pkg_id(pkg_id: str) -> Tuple[str, Optional[str]]:
     if "@" not in pkg_id:
         return pkg_id, None
     pkg_name, safe_pkg_version = pkg_id.split("@")
```

Comment:
This call still jumps out to me as a minor problem.  It relies on validation that happens outside of this function.  If you change this to `pkg_name, _, safe_pkg_version = pkg_id.partition('@')` then this code would be slightly more robust.  Totally a *nitpick* on my part.